File: practical03/integral.c
#include <stdio.h>
#include <math.h>

float main(void) {

/* Declaring variables */
    int N, i;
    float a, b, f, x, calc;

/* Start point, end point, points to end */
    a = 0;
    b = M_PI / 3;
    N = 12;

/* Adding start and end height to calculation */

    calc += tan(a);
    calc += tan(b);

/* Setting first point after start, for loop to add trapezoid heights between start and end */
    x = a + ( (b - a) / N);
    for (i=1; i<N; i++) {
        calc += (2 * tan(x));
        x += ( (b - a) / N);
    }

/* Multiplying by width of trapezoids */

    calc = calc * ( (b - a) / (2 * N));

/* Printing calculated value, analytical value for comparision */
    printf(" The value calculated using the trapezoidal rule is: %.4f\n", calc);
    printf(" The comparison value is: %.4f\n", log(2));

}

File: practical03/integral.f90
program integral
! Declare variables and parameters
    integer :: i, N
    real :: a,b,x,calc
    real, parameter :: pi = 3.1415927

! Setting start and end points, # of points to end
    a = 0
    b = pi / 3
    N = 12

! Adding inital and final trapezoid heights
    calc = tan(a)
    calc = calc + tan(b)

! Initalizing first end point, for loop to add heights
    x = a + ((b - a) / N)
    i = 1
    do while (i < N)
        calc = calc + (2*tan(x))
        x = x + ( (b - a) / N )
        i = i+1
    end do

! Multiplying by trapezoid widths
    calc = calc * ((b - a) / (2*N))

! Writing calculated value, analytical value for comparison
    write(6, '(a,f0.4)') ' The value calculated using the trapezoidal rule is: ',calc
    write(6, '(a,f0.4)') ' The value calculated analytically is:',log(2.)

end program integral
